tensorrt_llm/_torch/auto_deploy/custom_ops/flashinfer_attention.py

```diff
@@ -82,11 +85,45 @@ def init_workspace(self, workspace_buffer: torch.Tensor):
         self.decode_wrapper = self._init_decode_wrapper()
 
     def reset(self) -> None:
-        self.plan_params = None
+        self.plan_params_prefill = None
+        self.plan_params_decode = None
 
-    def plan(
+    def plan_prefill(
+        self,
+        qo_indptr_host: torch.Tensor,
+        kv_page_indptr_host: torch.Tensor,
+        kv_page_indices: torch.Tensor,
+        kv_last_page_len_host: torch.Tensor,
+        kv_lens_arr_host: torch.Tensor,
+        seq_len_host: torch.Tensor,
```

Comment:
we use host versions here. the `plan` actually needs both (host and device) version. Unfortunately, there is no good way to access the plan API and provide both although we have both available.

I have decided to use the host versions here to ensure non-blocking invocation of `plan`, whereas the other way around would trigger a blocking copy to cpu. This way we trigger a non-blocking copy to device (note that this is safe since we do have pinned CPU memory for all our host-side arguments).

TODO: document this in the code